composer.json

```diff
@@ -71,5 +71,11 @@
             "webroot/themes/contrib/{$name}": ["type:drupal-theme"],
             "webroot/sites/all/drush/contrib/{$name}": ["type:drupal-drush"]
         }
+    },
+    "config": {
+        "platform": {
+            "php": "7.1.18",
```

Comment:
@subtera745 @jackinloadup is this the exact version that's needed? We need to be this explicit? I feel like this might get away from us if the alpine container updates the version for the dev container.

Reply:
@smfsh Like you said, symfony required PHP 7.1. Well it actually required  7.1.3. I figured we would set it to the set it to the current version delivered by our alpine image ```php:7.1-fpm-alpine``` at this point. Looks like there isn't a way of stating 7.1.x or something in the [platform property](https://getcomposer.org/doc/06-config.md#platform).

Thoughts?

Reply:
@jackinloadup I understand the idea, but don't you think this introduces too much code debt? We'll need to change this platform directive _every_ time the alpine container is updated, which we don't control, so therefor we don't track. This is gonna be out of sync fast.

The patch version (18 in the case above) really should be non-breaking to the API, theoretically speaking. I think we'd be safe to set the line to `"php": "7.1",`. If a developer has a package requiring a specific patch version, I'd be really leery and want the end user to be cautioned about it anyway.

Reply:
@smfsh I wish settings it to ```7.1``` worked, but it doesn't. I was thinking we could set it specifically to ```7.1.3``` which would in essence say "we support AT LEAST ```7.1.3```". If we are going to do that we might as well use the specific version that we are on in the sense that our alpine images are not going backward. The out of date thing would happen and would at some point be an issue. Granted we might switch to ```7.2``` by then and we might be able to not specify patch versions.

A possible solution might be to script on docker-rebuild setting that value in the composer.json file with sed. Not a huge fan but actually not terrible. It would be more accurate. It could mean that ```make init``` or others would actually cause a clean repo to become dirty if alpine updated their image. I don't think there is anything else that makes the repo dirty when building.

Other thoughts or ideas? I understand the concerns you expressed. Besides above I'm not sure what other options are out there. I still think using this method is way better than some of the other ideas we talked about. Like putting composer into the php container.